modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/GridCoverage2DReader.java

```diff
@@ -273,6 +275,7 @@
      * 
      * @return The number of predetermined overviews for the default coverage. Zero if none are available, -1 if infinite are available, otherwise a
      *         positive number.
+     * @deprecated It should be used getDatasetLayout() if defined
```

Comment:
"Use getDataSetLayout().getNumOverviews()."
(or whatever the name of the overview fetching method is).

What does "if defined" mean. getNumOverviews is always availabe, getDatasetLayout should be equally always available.